pkg/controller/operators/olm/operator.go

```diff
@@ -392,13 +407,71 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 		return nil, err
 	}
 
+
+	// setup proxy env var injection policies
+	discovery := config.operatorClient.KubernetesInterface().Discovery()
+	proxyAPIExists, err := proxy.IsAPIAvailable(discovery)
+	if err != nil {
+		op.logger.Errorf("error happened while probing for Proxy API support - %v", err)
+		return nil, err
```

Comment:
this shouldn't error out if we're not on an openshift cluster, right?

Reply:
It should not, `err` is set only when there is a connectivity error. check - https://github.com/operator-framework/operator-lifecycle-manager/pull/950/commits/35a159ed8e1b1d7931a09f93a68a7c74c90540ff#diff-eb26d569f984271521cd371191e0b72bR30